### PR TITLE
[WM-2023] Migrate Find a Workflow

### PR DESCRIPTION
**Pull request by kpierre13**
## Summary of changes:

Adding functionality to the Find Workflow Button behind a feature flag. To enable the flag, please run `configOverridesStore.set({ isFindWorkflowEnabled: true, isDockstoreEnabled: true, isURLEnabled: true,  isAzureWorkflowsTabVisible: true })` from the Chrome console.

### What
- Adding in functionality for the `Find a Workflow` button

### Why
- Part of the migration of CBAS-UI into Terra-UI.

### Testing strategy
- Tests added as part of the PR.

 

https://github.com/DataBiosphere/terra-ui/assets/68349264/3e555859-595f-4c23-9c71-a558f391fa59

[WM-2023]: https://broadworkbench.atlassian.net/browse/WM-2023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/workflows-app/components/FindWorkflowModal.js [hunk 1]
```diff
@@ -1,16 +1,158 @@
...
+const styles = {
+  findWorkflowSubHeader: (selected) => {
+    return {
+      ...Style.navList.itemContainer(selected),
+      ...Style.navList.item(selected),
+      ...(selected ? { backgroundColor: colors.accent(0.2) } : {}),
+      paddingLeft: '3rem',
+    };
+  },
+};
```
Comment (dillydally414) [inline src/workflows-app/components/FindWorkflowModal.js:27]: I'm getting some axe accessibility issues in the console when opening the modal

File: src/libs/ajax/ajax-common.ts [hunk 2]
```diff
@@ -228,6 +228,11 @@ export const fetchEcm = _.flow(
   withRetryAfterReloadingExpiredAuthToken
 )(fetchOk);
 
+export const fetchCbas = _.flow(
+  withUrlPrefix(`${getConfig().cbasUrlRoot}/api/batch/v1/`),
+  withRetryAfterReloadingExpiredAuthToken
+)(fetchOk);
+
```
Comment (dillydally414) [inline src/libs/ajax/ajax-common.ts]: Is this used anywhere? In #3986 we decided to replace this with a generic fetchFromProxy so as to not duplicate across Cromwell and CBAS api calls routed through a proxy.

**Comment by salonishah11** [comment]
@kpierre13 it looks like it is importing commits that are already merged to dev. Maybe this branch needs a rebase with dev branch?

File: src/pages/ImportWorkflow/ImportWorkflow.js [hunk 3]
```diff
@@ -85,9 +85,10 @@ export const ImportWorkflow = ({ path, version, source }) => {
...
       };
       const res = await Ajax(signal).Cbas.methods.post(appUrls.cbasUrl, postRequestBody);
+      const methodId = res.method_id;
 
       await setAzureCookieOnUrl(signal, appUrls.cbasUiUrl, true);
-      window.location = `${appUrls.cbasUiUrl}#submission-config/${res.method_id}`;
+      Nav.goToPath('workspace-workflows-app-submission-config', { name, namespace, methodId });
     } else {
       throw new Error(
         'Error identifying a unique and valid Cromwell App. Cromwell Apps are valid if they were created by the current user in the workspace and are in a Running state.'
```
Comment (dillydally414) [inline src/pages/ImportWorkflow/ImportWorkflow.js:94]: Minor, are we rebranding this as "Workflows App"

File: src/workflows-app/components/FindWorkflowModal.js [hunk 4]
```diff
@@ -1,16 +1,158 @@
...
+  };
+
+  const isSubHeaderActive = (subHeader) => selectedSubHeader === subHeader;
 
-const FindWorkflowModal = ({ onDismiss }) => {
   return h(
     ModalDrawer,
     {
-      'aria-label': 'Find a Workflow Modal',
+      'aria-label': 'find-workflow-modal',
```
Comment (dillydally414) [inline src/workflows-app/components/FindWorkflowModal.js:88]: We should leave this label the same as per [this comment](https://github.com/DataBiosphere/terra-ui/pull/3986#discussion_r1244289288)

**Comment by kpierre13** [comment]
@dillydally414 I think a lot of your comments are due to something weird happening when rebasing and merging. I'll make sure to go through and try to correct them all.

File: src/pages/ImportWorkflow/ImportWorkflow.js (hunk 3 above)
Comment (kpierre13) [reply src/pages/ImportWorkflow/ImportWorkflow.js:94]: So: `workflows-app-submission-config` ?

**Comment by cahrens** [comment]
@kpierre13 your PR is showing code changes for things that are already in dev. Can you rebase (and mark this draft if it's not ready for review)?

**Pull request closed without merging**